Create asterisk.png without digest after precompile

SuperTux88 opened the pull request:
Fixes #7317

SuperTux88 committed:
Create asterisk.png without digest after precompile

fixes #7317

jhass commented:
Shouldn't we rather fix linking to a non-digest version?

svbergerem commented:
Like #7317 mentions this is linked as the OpenGraph image:

```
<meta property="og:image" content="/assets/branding/logos/asterisk.png" />
```

Are the old digest versions still available after creating a few new ones? We only look up the image once in our own implementation.

jhass commented:
Yes, they're kept until you run `rake assets:clean`.

svbergerem commented:
https://wiki.diasporafoundation.org/Updating#Post_update_cleanup

jhass commented:
Yeah, I'm aware, I'm just not too much of fan of side-stepping the cache bursting mechanism here. Perhaps we can rather define some routes with temporary redirects to the digest version?

jhass commented on lib/tasks/assets.rake:
```diff
@@ -10,9 +10,32 @@ namespace :assets do
+    logger = ::Logging::Logger["assets:non_digest_assets"]
+
+    non_digest_assets = Diaspora::Application.config.assets.non_digest_assets
+    manifest_path = Dir.glob(File.join(Rails.root, "public/assets/manifest-*.json")).first
```
This feels rather fragile, are we sure there's never more than one manifest?

SuperTux88 replied:
I only have one file, and it gets updated when I do a precompile. If somebody has two files, it's probably a problem anyway, because sprocket also simply takes the first file: https://github.com/rails/sprockets/blob/v2.12.4/lib/sprockets/manifest.rb#L47-L55

SuperTux88 commented:
The asterisk.png didn't change since two year, so it still is the same digest version. But every cached opengraph-result now links to the non-digest version.

I copied it once by hand to create the non-digest version, but that doesn't help other pods (and also it isn't there anymore, so I deleted it once by accident, probably when updating to 0.6, because I deleted the whole assets folder to have a clean start).

There is also a gem doing redirects from the non-digest versions to the latest digest version, but then every time somebody loads it, it needs two requests.

skyschub reviewed (approved):
Given the fact that this image is referenced externally and the digest-version could disappear, I am fine with using the non-digest version here!

svbergerem commented:
Thank you :)
